aliyun_driver.go

```diff
@@ -540,6 +545,10 @@ func (d *Driver) waitAliyunCluster(ctx context.Context, svc *cs.Client, state *s
 			return fmt.Errorf("aliyun failed to provision cluster: %s", status)
 		}
 		time.Sleep(time.Second * 15)
+
+		if startTime.Add(time.Duration(timeoutMins) * time.Minute).Before(time.Now()) {
+			return fmt.Errorf("timeout waiting cluster %q to be running", state.Name)
+		}
```

Comment:
Would the logic below for timeout will make the code more readable? Thanks.
``` golang
	timeout := time.After(30 * time.Minute)
	tick := time.Tick(15 * time.Second)
	// Keep trying until we're timed out or got a result or got an error
	for {
		select {
		// Got a timeout! fail with a timeout error
		case <-timeout:
			return fmt.Errorf("timedout waiting cluster ....")
		// Got a tick, we should check on cluster state and return
		case <-tick:
			cluster, err := getCluster(svc, state)
                         // return something
		}
	}
```

Reply:
Updated